codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2StreamChannel.java

```diff
@@ -934,24 +932,22 @@ public void write(Object msg, final ChannelPromise promise) {
         }
 
         private void writeHttp2StreamFrame(Http2StreamFrame frame, final ChannelPromise promise) {
-            if (!firstFrameWritten && !isStreamIdValid(stream().id()) && !(frame instanceof Http2HeadersFrame)) {
+            // If this stream doesn't have a stream ID it is because this stream is in the IDLE
+            // state. Note that we use the stream ID because the stream ID gets assigned right away
+            // while the underlying Http2Stream may get set later if the stream is queued due to
+            // MAX_CONCURRENT_STREAMS settings or similar reasons.
+            final boolean openingFrame = !isStreamIdValid(stream().id());
```

Comment:
If the `write0` call gets bounced through the executor I think this can cause issues, so maybe this is a no-go.

Reply:
hmm `write0` should only be called from within the EventLoop. What problems you are seeing here ?

Reply:
or you talking that the stream id may be valid directly after the write0 returns and so it may be problematic. If you  talk about this then I agree, with means this PR may cause trouble

Reply:
Yes, that is what I was worried about. I'll drop this.